src/Features/Core/Portable/UseCoalesceExpression/AbstractUseCoalesceExpressionDiagnosticAnalyzer.cs

```diff
@@ -95,12 +97,30 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
             }
 
             if (!syntaxFacts.AreEquivalent(
-                    conditionRightIsNull ? conditionLeftLow : conditionRightLow, 
+                    conditionRightIsNull ? conditionLeftLow : conditionRightLow,
                     isEquals ? whenFalseNodeLow : whenTrueNodeLow))
             {
                 return;
             }
 
+            var semanticModel = context.SemanticModel;
+            var conditionType = semanticModel.GetTypeInfo(
+                conditionLeftIsNull ? conditionRightLow : conditionLeftLow, cancellationTokan).Type;
+            if (conditionType != null &&
```

Comment:
:question: Why not use the following condition here?

```csharp
if (conditionType == null || !conditionType.IsReferenceType)
```

It seems like if `conditionType` is null then we can't be sure the operation is safe and should probably skip it.